File: BackEnd/src/domain/interfaces/usecases/community/ICreateCommentUseCase.ts
/**
 * @file ICreateCommentUseCase.ts
 * @description Interfaz del caso de uso para crear un comentario en un post.
 * @module Community
 * @layer Domain
 */

import type { CommentResponseDTO } from '../../../dtos/community/comment-response.dto.js';

/**
 * Contrato del use case CreateComment (alias: AddComment en types.ts).
 */
export interface ICreateCommentUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param postId — Id del post en el que se comenta.
   * @param authorId — Id del usuario que escribe el comentario.
   * @param content — Texto del comentario (máx. 500 chars).
   * @returns CommentResponseDTO del comentario creado.
   * @throws {NotFoundException} Si el post no existe o está eliminado.
   */
  execute(postId: string, authorId: string, content: string): Promise<CommentResponseDTO>;
}

File: FrontEnd/lib/domain/interfaces/usecases/community/i_create_comment_use_case.dart
/// @file i_create_comment_use_case.dart
/// @description Interfaz: Crea un comentario en un post.
/// @module Community
/// @layer Domain
library;

import '../../../entities/comment.dart';

abstract interface class ICreateCommentUseCase {
  /// Crea un comentario en un post.
  ///
  /// [param] postId  — Identificador del post.
  /// [param] content — Texto del comentario.
  /// [returns] [Comment] recién creado con datos del servidor.
  Future<Comment> execute(String postId, String content);
}
